designs/compatibility/arm64-support.md

```diff
@@ -0,0 +1,150 @@
+
+- **Control-plane Builds**:
+  - Update control-plane builds to support `arm64`.
+
+- **Data-plane build**:
+  - Update data-plane builds to support `arm64`.
+    - LocalPV-HostPath
+    - LocalPV-ZFS
+    - LocalPV-LVM
+    - ReplicatedPV-Mayastor
+  
+- **Testing Strategy**:
+  - Introduce `arm64` test nodes for integration tests.
+  - Run key conformance tests and functional tests on `arm64` platforms to ensure feature parity.
+
+- **Documentation and Examples**:
+  - Update the official OpenEBS docs to include instructions for deploying on `arm64` clusters.
+  - Provide example manifests and Helm chart values showing how to install and verify `arm64` deployments.
+
+### Risks and Mitigations
+
+- **Risk: CI Complexity**: Adding `arm64` pipelines may increase build time and complexity.
+  - *Mitigation*: Use cloud-based ARM runners or Arm-enabled CI platforms, and possibly maintain separate test schedules for `arm64` to reduce load.
+  
+- **Risk: Limited Community Hardware**: Some maintainers and community members might not have immediate access to native `arm64` hardware for debugging.
+  - *Mitigation*: Utilize public cloud credits for temporary Arm instances or rely on well-supported emulation to replicate issues.
+
+- **Risk: External Dependencies**: Some upstream dependencies may lack `arm64` builds.
+  - *Mitigation*: Work closely with dependency maintainers or find equivalents that support `arm64`. Document any known gaps or compatibility issues.
+
+## Graduation Criteria
+
+- **Alpha (Initial Support)**:
+  - `arm64` images are built and published for all primary OpenEBS components.
+  - Basic integration tests pass on an `arm64` test cluster.
+  - Documentation updated to reflect `arm64` availability.
+
+- **Beta (Expanded Support)**:
+  - Comprehensive integration tests and end-to-end tests run regularly on `arm64`.
+  - Users have reported successful deployments without critical issues.
+  - Helm charts and manifests verified for `arm64` environments.
+
+- **General Availability (GA)**:
+  - `arm64` support is included in stable OpenEBS releases.
+  - Formal support commitments are documented (e.g., backporting critical fixes).
+
+## Implementation History
+
+- **2024-12-13**: Draft OEP with summary and motivation created (provisional status).
+
+## Drawbacks [optional]
+
+- Increased continuous integration resources and compute time.
+- Additional maintenance overhead and potential for new `arm64`-specific issues.
+- Broader testing matrix may slow down release cadence.
+
+## Alternatives [optional]
+
+- Continue supporting only `amd64`.
+- Partial support with unofficial or best-effort builds without rigorous testing.
+- Wait for broader industry support and proven `arm64` tooling maturity before investing in this effort.
+
+## Infrastructure Needed [optional]
+
+- `arm64` CI runners or cloud-based environments for testing.
```

Comment:
We are unable to add more users to the current mayastor Jenkins.
So we need an item here to fix that, either of these:
- move from Jenkins to github actions 
- setup Jenkins on cncf hardware and migrate to it 

I suspect the github actions free runners are not bulky enough to test mayastor, but we do have some CNCF github actions runners, these may be good enough? 
CC @Abhinandan-Purkait @niladrih

Reply:
I'm not sure how the current system is setup, as I haven't dug into it quite yet, but I'll try to diagram it out as I learn the setup and then add docs on how `arm64` testing plays into that.

Reply:
Btw, started further improving the build&test docs: https://github.com/openebs/mayastor/pull/1783

Reply:
Yeah, we can give the cncf ones a try and they also have arm machines iirc.